websocket/client.go

```diff
@@ -470,13 +471,16 @@ func (c *Client) Connect() error {
 	var err error
 	endpoint := c.URL.String()
 	headers := http.Header{
-		"X-Host-ID":             {c.Host},
 		"Origin":                {endpoint},
 		"X-Client-Type":         {c.ServiceType.String()},
 		"X-Client-Protocol":     {c.ClientProtocol.String()},
 		"X-Websocket-Namespace": {WildcardNamespace},
 	}
 
+	if !c.Opts.NoHostID {
```

Comment:
what kind of issue or use case are you trying to address ? can you elaborate ?

Reply:
BTW host_id are supposed to be unique, that is the purpose of this ID

Reply:
The use case is VPC logs in which we want to run multiple instances of a websocket listener on the same host, each instance supporting a different tenant. So I guess if not explicitly set all would resolve to the same host_id and thus the 2nd client would conflict - so when running my multi client tests w/o removing X-Host-ID field from client request I got:

```
[vagrant@dev websocket]$ vim client.go
[vagrant@dev websocket]$ vim server_test.go
[vagrant@dev websocket]$ go test
2019-06-04T08:56:59.795Z        INFO    http/server.go:104 (*Server).Listen     dev Listening on socket localhost:59999
2019-06-04T08:56:59.795Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:56:59.796Z        INFO    websocket/server.go:118 (*Server).newIncomingClient     dev New WebSocket Connection from 127.0.0.1:53652 : URI path /wstest
2019-06-04T08:56:59.797Z        INFO    websocket/client.go:512 (*Client).Connect       dev Connected to ws://localhost:59999/wstest
2019-06-04T08:57:01.796Z        INFO    http/server.go:104 (*Server).Listen     dev Listening on socket localhost:59999
2019-06-04T08:57:01.796Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:01.797Z        INFO    websocket/server.go:118 (*Server).newIncomingClient     dev New WebSocket Connection from 127.0.0.1:53656 : URI path /wstest
2019-06-04T08:57:01.797Z        INFO    websocket/client.go:512 (*Client).Connect       dev Connected to ws://localhost:59999/wstest
2019-06-04T08:57:03.799Z        INFO    http/server.go:104 (*Server).Listen     dev Listening on socket localhost:59999
2019-06-04T08:57:03.800Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.801Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.802Z        INFO    websocket/server.go:118 (*Server).newIncomingClient     dev New WebSocket Connection from 127.0.0.1:53662 : URI path /wstest
2019-06-04T08:57:03.803Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53664) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.803Z        INFO    websocket/client.go:512 (*Client).Connect       dev Connected to ws://localhost:59999/wstest
2019-06-04T08:57:03.803Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.803Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.803Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.803Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.804Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.804Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53672) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.804Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53670) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.804Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.804Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.805Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.805Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.805Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.805Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:03.806Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53676) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.806Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53696) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.806Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53688) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.806Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53690) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.807Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53692) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.807Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53694) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:03.807Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.807Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.807Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.807Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.807Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:03.808Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.803Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.805Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53700) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.805Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.806Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.806Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53706) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.807Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53708) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.807Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.808Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.808Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.808Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.809Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.809Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.810Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.810Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.812Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:04.812Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53722) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.812Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53718) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.813Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53720) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.814Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.814Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.814Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.814Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53728) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.815Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53726) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.815Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.815Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:04.816Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53732) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:04.816Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.808Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.809Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.811Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53740) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.811Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53736) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.813Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.813Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.814Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.814Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53744) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.814Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.815Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.816Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.816Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.816Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.816Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.816Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:05.816Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53752) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.817Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.817Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53754) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.817Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.817Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53756) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.817Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.818Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53760) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.818Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53768) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.818Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.818Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53766) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:05.818Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:05.819Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.814Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.815Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.816Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.818Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53772) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.818Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.819Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.820Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.821Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.822Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53784) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.822Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53780) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.822Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53782) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.822Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.823Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.823Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.823Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.823Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.823Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:06.824Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53794) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.824Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53792) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.824Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53796) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.825Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.825Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.825Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.825Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53802) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.825Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53804) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:06.826Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:06.826Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.823Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.824Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.825Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53808) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.825Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.825Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.825Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.825Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53812) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.825Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.826Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.826Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.826Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.827Z        INFO    websocket/client.go:488 (*Client).Connect       dev Connecting to ws://localhost:59999/wstest
2019-06-04T08:57:07.827Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.828Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53820) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.828Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53822) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.828Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.828Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.828Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53834) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.828Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53836) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.828Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53828) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.828Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.829Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.829Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53838) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.829Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.829Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
2019-06-04T08:57:07.829Z        ERROR   websocket/server.go:87 (*Server).serveMessages  dev host_id 'myhost' (127.0.0.1:53840) conflicts, same host_id used by &{Conn:0xc420159040}
2019-06-04T08:57:07.829Z        ERROR   websocket/client.go:550 (*Client).Start.func1   dev Unable to create a WebSocket connection ws://localhost:59999/wstest : websocket: bad handshake
--- FAIL: TestSubscriptionMultiClient (5.01s)
        server_test.go:280: Client should have received 2 OnConnected events: 0
FAIL
exit status 1
```